### PR TITLE
ci: Add python 3.11 and minor CI fixing

### PR DESCRIPTION
**Pull request by georgebisbas**

### CODE REVIEW

File: docker/Dockerfile.cpu [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # This Dockerfile contains the Devito codes and can be built using different base images.
 ##############################################################
 
-ARG pyversion=python:3.7
+ARG pyversion=python:3.8
```
Comment (EdCaunt) [inline docker/Dockerfile.cpu:5]: Why not get ahead of the curve and use Python 3.9 or 3.10 here (and elsewhere)? Looks like the GPU one uses 3.9

File: .github/workflows/pytest-core-nompi.yml [hunk 2]
```diff
@@ -31,14 +31,14 @@ jobs:
...
            pytest-ubuntu-py310-gcc10-noomp,
            pytest-ubuntu-py38-gcc8-omp,
            pytest-ubuntu-py39-gcc9-omp,
-           pytest-osx-py37-clang-omp,
-           pytest-docker-py37-gcc-omp,
-           pytest-docker-py37-icc-omp,
+           pytest-osx-py38-clang-omp,
+           pytest-docker-py38-gcc-omp,
+           pytest-docker-py38-icc-omp,
            pytest-docker-py38-icx-omp
```
Comment (EdCaunt) [inline .github/workflows/pytest-core-nompi.yml:42]: If the others are bumped up a version, would it make sense to do the same for this one (3.8->3.9)?

File: .github/workflows/pytest-core-nompi.yml [hunk 3]
```diff
@@ -50,15 +50,15 @@ jobs:
...
-        - name: pytest-ubuntu-py38-gcc12-omp
-          python-version: '3.8'
+        - name: pytest-ubuntu-py311-gcc12-omp
+          python-version: '3.11'
           os: ubuntu-22.04
           arch: "gcc-12"
           language: "openmp"
           sympy: "1.10"
 
         - name: pytest-ubuntu-py39-gcc7-omp
```
Comment (georgebisbas) [inline .github/workflows/pytest-core-nompi.yml:60]: tofix

File: docker/Dockerfile.cpu (hunk 1 above)
Comment (georgebisbas) [reply docker/Dockerfile.cpu:5]: I am open to this, let's see what others think

File: .github/workflows/pytest-core-nompi.yml (hunk 2 above)
Comment (georgebisbas) [reply .github/workflows/pytest-core-nompi.yml:42]: I should have another look to "keep some balance for all"

File: .github/workflows/examples.yml [hunk 4]
```diff
@@ -87,7 +87,7 @@ jobs:
         python examples/cfd/example_diffusion.py
 
     - name: Upload coverage to Codecov
-      if: matrix.name != 'pytest-docker-py37-gcc-omp'
+      if: matrix.name != 'pytest-docker-py38-gcc-omp'
```
Comment (mloubout) [inline .github/workflows/examples.yml:90]: This isn't right, there is no pytest or matrix name in this workflow it's only run once, if should be removed

File: docker/Dockerfile.cpu (hunk 1 above)
Comment (mloubout) [reply docker/Dockerfile.cpu:5]: 3.9 would be a safe option yes

**Review by mloubout — COMMENTED** [review]
What's going on with all those notebook changes

**Comment by georgebisbas** [comment]
> What's going on with all those notebook changes

There is some different latex printing adding this \operatorname

File: devito/core/autotuning.py [hunk 5]
```diff
@@ -122,7 +122,8 @@ def autotune(operator, args, level, mode):
             at_args.update(dict(run))
 
             # Drop run if not at least one block per thread
-            if not configuration['develop-mode'] and nblocks_per_thread.subs(at_args) < 1:
+            if not configuration['develop-mode'] and \
```
Comment (georgebisbas) [inline devito/core/autotuning.py:125]: Fix for Sympy1.12 from @mloubout

File: devito/tools/utils.py
Comment (georgebisbas) [inline devito/tools/utils.py:1]: not related to this PR, but minor addition

File: .github/workflows/examples.yml [hunk 6]
```diff
@@ -86,7 +86,7 @@ jobs:
         python examples/cfd/example_diffusion.py
 
     - name: Upload coverage to Codecov
-      if: matrix.name != 'pytest-docker-py37-gcc-omp'
+      if: matrix.name != 'pytest-docker-py39-gcc-omp'
```
Comment (mloubout) [inline .github/workflows/examples.yml:89]: Leftover delete line

**Review by mloubout — COMMENTED** [review]
I wouldn't change the tutorials it's unnecessary diff

**Review by mloubout — APPROVED** [review]

**Review by FabioLuporini — APPROVED** [review]

**Pull request merged**